Fix tuple pattern warning related with xcode upgrade to 11.4

mauriciomaniglia opened the pull request:
This warning appears after upgrading to Xcode 11.4 

https://developer.apple.com/documentation/xcode_release_notes/xcode_11_4_release_notes

`Code that relies on the compiler automatically tupling a pattern may lead to a compiler error when upgrading to Xcode 11.4, even though the code compiled before. (58425942)`

mauriciomaniglia committed:
Fix tuple pattern warning related with xcode upgrade to 11.4

https://developer.apple.com/documentation/xcode_release_notes/xcode_11_4_release_notes

`Code that relies on the compiler automatically tupling a pattern may lead to a compiler error when upgrading to Xcode 11.4, even though the code compiled before. (58425942)`

caiozullo commented on EssentialFeed/EssentialFeedTests/Feed API/URLSessionHTTPClientTests.swift:
```diff
@@ -92,8 +92,8 @@ class URLSessionHTTPClientTests: XCTestCase {
-		case let .success(data, response):
-			return (data, response)
+		case let .success(successResult):
+			return successResult
```
Just a naming suggestion: The name of the function is `resultValuesFor(...)` so I'd expect to return 'values' (instead of `successResult`)

```suggestion
		case let .success(values):
			return values
```

caiozullo reviewed (commented):
Hi @mauriciomaniglia, thank you for the PR! 

This is an annoying Xcode issue - hopefully, they'll fix it in the next version.

I added a suggestion to improve a name - please, have a look.

mauriciomaniglia committed:
Update return value name to reflect the name of the function.

Update return value name, since this function returns "values" not a "success result".

Co-authored-by: Caio Zullo <caiozullo@gmail.com>

mauriciomaniglia commented:
@caiozullo Done! What do you think about the commit message ?

caiozullo commented:
Looks good, @mauriciomaniglia - thanks again!
